src/selectors/chain.ts

```diff
@@ -1,19 +1,26 @@
 import { getContextsSortedAndRanked } from '../selectors'
-import { head, splice } from '../util'
-import { Path } from '../types'
+import { head, pathToContext, splice } from '../util'
+import { Context, Path } from '../types'
 import { State } from '../util/initialState'
 
+/** Return true if superContext includes subContext. */
+const isContextIncludes = (superContext: Context, subContext: Context): boolean => {
+  const superAsString = superContext.join('/')
+  const subAsString = subContext.join('/')
+  return superAsString.includes(subAsString)
+}
+
 /** Merges thoughts into a context chain, removing the overlapping head. */
 // use autogenerated rank of context
 // if there is no/empty context chain, return thoughtsRanked as-is
-const chain = (state: State, contextChain: Path[], path: Path) => {
+const chain = (state: State, contextChain: Path[], path: Path, showContexts?: boolean) => {
```

Comment:
The head of each `Path` in the `contextChain` represents a Context View, so we shouldn't need an extra variable `showContext`. We should discuss our approach more however before attempting an implementation.